bin/gobblin-mapreduce.sh

```diff
@@ -14,26 +14,46 @@ fi
 
 JOB_TRACKER_URL=$1
 FS_URL=$2
-JOB_CONFIG_FILE=$3
+JOB_CONFIG_FILE=$4
+
+# Convert job jar's relative path to absolute path
+DIRNAME=`dirname "$3"`
+JARNAME=`basename "$3"`
+DIRNAME=`cd "$DIRNAME">/dev/null; pwd`
+JOB_JAR_FILE="$DIRNAME/$JARNAME"
 
 FWDIR="$(cd `dirname $0`/..; pwd)"
+FWDIR_LIB=$FWDIR/lib
 
 # Jars Gobblin runtime depends on
-LIBJARS=$FWDIR/lib/common-api.jar,$FWDIR/lib/metastore.jar,$FWDIR/lib/qualitychecker.jar,\
-$FWDIR/lib/source.jar,$FWDIR/lib/converter.jar,$FWDIR/lib/writer.jar,$FWDIR/lib/utility.jar,\
-$FWDIR/lib/guava-15.0.jar,$FWDIR/lib/avro-1.7.1.jar,$FWDIR/lib/avro-mapred-1.7.6.jar,\
-$FWDIR/lib/metrics-core-3.0.2.jar,$FWDIR/lib/gson-2.2.4.jar
-
-export HADOOP_CLASSPATH=$FWDIR/lib/*
+LIBJARS=$FWDIR_LIB/gobblin-metastore.jar,$FWDIR_LIB/gobblin-metrics.jar,\
+$FWDIR_LIB/gobblin-core.jar,$FWDIR_LIB/gobblin-api.jar,$FWDIR_LIB/gobblin-utility.jar,\
+$FWDIR_LIB/guava-15.0.jar,$FWDIR_LIB/avro-1.7.6.jar,$FWDIR_LIB/avro-mapred-1.7.6.jar,\
+$FWDIR_LIB/metrics-core-3.0.2.jar,$FWDIR_LIB/gson-2.2.4.jar,$FWDIR_LIB/joda-time-1.6.jar,$FWDIR_LIB/data-1.15.9.jar,\
+$JOB_JAR_FILE
+
+# Add libraries to the Hadoop classpath
+GOBBLIN_DEP_JARS=""
+for jarFile in `ls $FWDIR_LIB/*`
+do
+  GOBBLIN_DEP_JARS=${GOBBLIN_DEP_JARS}:$jarFile
+done
+
+# Honor Gobblin dependencies
+export HADOOP_USER_CLASSPATH_FIRST=true
```

Comment:
Hi Lorand,

The reason why I'm asking this is because the configuration key used for the same purpose is different between Hadoop 1 and Hadoop 2. So I am wondering if the same environment variable is used for both?